api/src/main/java/me/lokka30/treasury/api/economy/currency/Currency.java

```diff
@@ -1,236 +1,141 @@
+     * @param subscription The {@link EconomySubscriber} accepting the resulting {@link Double} that
+     *                     represents the deformatted amount of the formatted String.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    protected void setConversionCoefficient(double conversionCoefficient) {
-        this.conversionCoefficient = conversionCoefficient;
-    }
+    void deformat(@NotNull String formatted, @NotNull EconomySubscriber<Double> subscription);
 
     /**
-     * Get the starting balance of the currency.
-     * The player {@link UUID} is nullable, it should be specified if the starting balance
-     * concerns a {@link me.lokka30.treasury.api.economy.account.PlayerAccount}.
+     * Gets the starting balance of a specific player account for this currency.
      *
-     * @param playerUUID a UUID of the player account created. For global scenarios, specify `null`.
-     * @return the starting balance of the currency concerning specified player's UUID.
-     * @author lokka30, Geolykt, MrIvanPlays
+     * @param playerID The UUID of the player we are getting the starting balance for.
+     * @return The starting balance of the player for this currency.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    public double getStartingBalance(@Nullable UUID playerUUID) {
-        return startingBalance.apply(playerUUID);
-    }
+    double getStartingBalance(@Nullable UUID playerID);
 
     /**
-     * Gets a human-readable format of the balance.
-     * For example, '$1.50' or '1.50 dollars' or 'One dollar and fifty cents'.
+     * Used to translate an amount to a user readable format with the default precision.
      *
-     * @param amount to be formatted.
-     * @param locale of the formatted balance being requested.
-     * @return the human-readable format of the specified amount and locale.
-     * @author lokka30, MrIvanPlays
+     * @param amount The amount to format.
+     * @return The formatted text.
+     * @author creatorfromhell
      * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
      */
-    @NotNull
-    public String formatBalance(double amount, @NotNull Locale locale) {
-        Objects.requireNonNull(locale, "locale");
-        return balanceFormatter.apply(amount, locale);
-    }
+    String format(@NotNull Double amount);
 
+    /**
+     * Used to translate an amount to a user readable format with the specified amount of decimal places.
+     *
+     * @param amount The amount to format.
+     * @param precision The amount of decimal digits to use when formatting.
+     * @return The formatted text.
+     * @author creatorfromhell
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    String format(@NotNull Double amount, @NotNull Integer precision);
```

Comment:
`@NotNull Double` why not just a `double`?
`@NotNull Integer` why not just a `int`?

Reply:
This is based on the fact that Treasury utilizes @NotNull when primitive types cannot be null nor can they be compared to null in java.